Index: M5GFX/src/lgfx/boards.hpp
===================================================================
--- M5GFX.orig/src/lgfx/boards.hpp
+++ M5GFX/src/lgfx/boards.hpp
@@ -58,6 +58,8 @@ namespace lgfx // This should not be cha
     , board_M5PowerHub
     , board_M5DualKey
     , board_M5UnitPoEP4
+    , board_M5StampS3Bat
+    , board_M5StampP4
 
 /// external displays
     , board_M5AtomDisplay = 192
