Index: lv_binding_micropython/lv_conf.h
===================================================================
--- lv_binding_micropython.orig/lv_conf.h
+++ lv_binding_micropython/lv_conf.h
@@ -497,7 +497,7 @@ extern void mp_lv_deinit_gc();
 /*Optionally declare custom fonts here.
  *You can use these fonts as default font too and they will be available globally.
  *E.g. #define LV_FONT_CUSTOM_DECLARE   LV_FONT_DECLARE(my_font_1) LV_FONT_DECLARE(my_font_2)*/
-#define LV_FONT_CUSTOM_DECLARE
+#define LV_FONT_CUSTOM_DECLARE LV_FONT_DECLARE(AlibabaPuHuiTi_CN24) LV_FONT_DECLARE(AlibabaSans_JP24) LV_FONT_DECLARE(AlibabaSans_KR24)
 
 /*Always set a default font*/
 #define LV_FONT_DEFAULT &lv_font_montserrat_14
